src/main/java/space/obminyashka/items_exchange/service/AdvertisementService.java

```diff
@@ -24,10 +24,24 @@ public interface AdvertisementService {
     Page<AdvertisementTitleDto> findAllThumbnails(Pageable pageable);
 
     /**
-     * Find 12 random advertisements as thumbnails
+     * Find N random advertisements as thumbnails with filters
+     *
+     * @param amount amount of random advertisement
+     * @param advertisementId exist id of advertisement for ignoring
+     * @param subcategoryId exist id of subcategory for filtering
      * @return random 12 advertisement
      */
-    List<AdvertisementTitleDto> findRandom12Thumbnails();
+    List<AdvertisementTitleDto> findRandomNThumbnails(int amount, UUID advertisementId, Long subcategoryId);
+
+    /**
+     * Find 4 random advertisements with same subcategory without request advertisement
```

Comment:
Both your controllers' methods have 12 as the default value. There is a mistake somewhere